src/ResourceManager/Media/Commands.Media/MediaService/SetAzureRmMediaService.cs

```diff
@@ -0,0 +1,146 @@
+        public Hashtable Tags { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The storage accounts assosiated with the media account.")]
+        [ValidateNotNull]
+        public PSStorageAccount[] StorageAccounts { get; set; }
+
+        [Parameter(Mandatory = false,
+           HelpMessage = "Force to set media service without confirm.")]
+        public SwitchParameter Force { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            var mediaServiceParams = new RestMediaService();
+
+            if (Tags != null)
+            {
+                mediaServiceParams.Tags = Tags.ToDictionaryTags();
+            }
+
+            if (StorageAccounts != null)
+            {
+                // check storage accounts parameter
+                var primaryStorageAccounts = StorageAccounts.Where(x => x.IsPrimary.HasValue && x.IsPrimary.Value).ToArray();
+                if(primaryStorageAccounts.Count() != 1)
+                {
+                    throw new ArgumentException("StorageAccounts should have exactly one primary storage account");
+                }
+
+                var mediaService = MediaServicesManagementClient.MediaService.Get(ResourceGroupName, AccountName);
+                if (mediaService == null)
+                {
+                    throw new ArgumentException(string.Format(
+                        "MediaServiceAccount {0} under subscprition {1} and resourceGroup {2} doesn't exist",
+                        AccountName,
+                        SubscrptionName,
+                        ResourceGroupName));
+                }
+
+                var primaryStorageAccount = mediaService.StorageAccounts.FirstOrDefault(x => (x.IsPrimary.HasValue && x.IsPrimary.Value));
+                // there must be a primary storage account associated with the media service account
+                if (primaryStorageAccount == null)
+                {
+                    throw new Exception(string.Format(
+                        "MediaServiceAccount {0} under subscprition {1} and resourceGroup {2} has no primary storage account",
+                        AccountName,
+                        SubscrptionName,
+                        ResourceGroupName));
+                }
+
+                if (!string.Equals(primaryStorageAccount.Id, primaryStorageAccounts[0].Id, StringComparison.OrdinalIgnoreCase))
+                {
+                    throw new ArgumentException("Primary storage account cann't be changed");
+                }
+
+                mediaServiceParams.StorageAccounts = StorageAccounts.Select(x => x.ToStorageAccount()).ToList();
+            }
+
+            try
+            {
+                if (ShouldProcess(AccountName, string.Format(SetMediaServiceWhatIfMessage)))
+                {
+                    if (Force || ShouldContinue(string.Format(SetMediaServiceWarning, AccountName), ""))
```

Comment:
WHy do you need a Force parameter or a ShouldContinue?  ShouldProcess satisfies Confirmation best practies in this case.